PolicyServicePkg/PolicyService/DxeMm/PolicyCommon.c

```diff
@@ -311,15 +323,216 @@ CommonSetPolicy (
     Entry->AllocationSize = PolicySize;
     CopyMem (Entry->Policy, Policy, PolicySize);
     InsertTailList (&mPolicyListHead, &Entry->Link);
+    Status = EFI_SUCCESS;
   }
 
-  Status = InstallPolicyIndicatorProtocol (PolicyGuid);
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "%a: Failed to install notification protocol. (%r)\n", __FUNCTION__, Status));
-    goto Exit;
+  Events = POLICY_NOTIFY_SET | (Entry->Attributes & POLICY_ATTRIBUTE_FINALIZED ? POLICY_NOTIFY_FINALIZED : 0);
+  CommonPolicyNotify (Events, Entry);
+
+  if (Attributes & POLICY_ATTRIBUTE_FINALIZED) {
+    Status = InstallPolicyIndicatorProtocol (PolicyGuid);

```

Comment:
I am thinking through if any consumers would need or expect the finalized policy to be published prior to getting the notification. I don't think so, the general use case would be to get notified to use the finalized policy. And because the protocol doesn't have any functions behind it, this should be ok. Curious your thoughts?

Reply:
This was intentional, The rationale being that the consumers that will be dispatched should always be last in my opinion. This maintains the ability to override policies (even ones that are finalized if needed) in the callbacks who I view as more enlightened users.

Reply:
That makes sense, thanks.